### ISSUE TITLE
part_of relation missing from GO basic

### ISSUE THREAD

**Issue by ValWood** [geneontology/go-ontology#19954]
part_of relation missing between SNAP receptor and membrane fusion.

https://user-images.githubusercontent.com/7359272/92286477-15eac500-eeff-11ea-9c85-c509f13a23a2.png

https://github.com/pombase/pombase-chado/issues/791

**Comment by pgaudet** [comment]
@kltm @dougli1sqrd @balhoff 
Can you have a look ? This is weird, I see the relation in protégé, but not in today's snapshot go-basic.obo 

Thanks, Pascale

**Comment by ValWood** [comment]
@kimrutherford for info

**Comment by balhoff** [comment]
I think relations that cross GO aspects/namespaces are specifically filtered out of go-basic. There is a Perl script for this: https://github.com/geneontology/go-ontology/blob/af1c8e3e48dec31842814b975209759f10485898/src/util/filter-obo-for-standard-release.pl#L69-L73

This one is a part_of between a MF and a BP.

**Comment by pgaudet** [comment]
@ValWood 

go-basic does not contain relations that go across aspects, see http://geneontology.org/docs/download-ontology/

I dont think we want to change this. 

Thanks, Pascale

**Comment by ValWood** [comment]
Just  so I can explain to Kim - why is that?

We use GO basic, but this means that we will miss inferences. 
Should GO basic not be used?
I imagined   GO would want to keep inferences consistent whichever flavour of the ontology is used? Otherwise tools and  databases would report annotation numbers differently depending whether they use GO full or GO basic.

What should GO basic be used for? Does it have a special purpose (if it does not contain inter-ontology relationships it should come with a big disclaimer).

@cmungall

**Comment by ValWood** [comment]
Thinking about it some more we don't need these inter ontology relationships in GO basic.
 
BUT, the annotations should be generated by GO, and we (POmBase) pick them up and incorporate them into our annotation corpus.
So, we seem to be missing the  annotations to "membrane fusion" that should be inferred via the  inter-ontology link?

So I need to figure out if we are not including these annotations, or if some annotations are not being generated, or if there is a lag.
v

**Comment by ValWood** [comment]
1. the relation is not new
GO:0005484    SNAP receptor activity
2015-01-12 | Added | RELATION | part of GO:0061025 (membrane fusion)

2. Are the  MF-BP annotations only generated if there is not an existing annotation?
If so, we may not be seeing these because  there are PAINT annotations and we are not yet importing PAINT.
(we are waiting until  these fixes are through 
https://github.com/geneontology/go-annotation/issues?q=is%3Aissue+is%3Aopen+label%3A%22PAINT+annotation%22)

I suspect that might be  the case, if so we can close this ticket now because we will get the inferences soon.

**Comment by pgaudet** [comment]
> Are the MF-BP annotations only generated if there is not an existing annotation?

I think so. @cmungall @kltm 
can you confirm ?

**Comment by ValWood** [comment]
Hi   @cmungall @kltm

can you confirm that  an inferred annotation between MF and BP will only be made if there is no existing annotation? If so I can close this..

**Comment by cmungall** [comment]
> Just so I can explain to Kim - why is that?

Historic. There was historically there were lots of tools that baked in assumptions about no inter-ontology links.

I think we should revisit this. We can give people the script.

**Comment by ValWood** [comment]
Hi @cmungall  
It's OK for us that the inter ontology links are absent from GO basic, because the annotations are provided so we will get them anyway. I think the only reason we don't have the annotation I was looking at is because we are not importing PAINT right now (we are waiting on some annotation fixes). So, if something has a PAINT (or any annotation from another source)    you don't create an annotation by a MF-BO link, is that correct?

**Comment by ValWood** [comment]
We are now importing PAINT ans still missing these annotation.

I can't figure out why.


13 | membrane fusion (GO:0061025) AND SNAP receptor activity (GO:0005484) [details] | ...
  | 34 | membrane fusion (GO:0061025) | ...
  | 19 | SNAP receptor activity (GO:0005484)

The intersect should be 19 based on the ontology

**Comment by ValWood** [comment]
Based on the ontology, anything annotated to 
"SNAP receptor activity"
should get an infered annotation to "membrane fusion"

https://www.ebi.ac.uk/QuickGO/GTerm?id=GO:0005484

Is the inferred annotation from F-P links up and running?

**Comment by pgaudet** [comment]
As thought the interontology links were used to make annotations that go in the 'prediction' files, for example for pombase: http://release.geneontology.org/2020-12-08/products/annotations/pombase-prediction.gaf

 But I dont see SPCC594.06c (one of the pombase genes annotated with SNAP receptor activity, see http://amigo.geneontology.org/amigo/gene_product/PomBase:SPCC594.06c) 

Not sure what's going on

@kltm Do you know how the pipeline handle those ? 

Thanks, Pascale

**Comment by ValWood** [comment]
The SNAP receptor activity for SPCC594.06c is ISO. Perhaps the F-P inferences ignores this evidence code? but it should be evidence code agnostic, if the annotation is true, the inference is true whatever the evidence code...

**Comment by ValWood** [comment]
This  old ticket documents some other instances of the same problems.
Quite often  when I expect that we did not need to make a process annotation be3casue it could be inferred, the expected annotation is not in the inferred gaf. This makes it difficult to know whether we need to make the gap filling annotations or not...

**Comment by ValWood** [comment]
This old ticket documents some other instances of the same problems.
Quite often, when I expect that we did not need to make a process annotation because it should be inferred via an F-P link, the expected annotation is missing from the inferred GAF. This makes it difficult to know whether we need to make the gap filling annotations or not...

see also
geneontology/go-annotation#1336
geneontology/go-annotation#1427

**Comment by pgaudet** [comment]
Maybe we can set up a call also with @kimrutherford and @balhoff to look at that ? I dont understand where the issue is coming from.

**Comment by ValWood** [comment]
I don't think it is an ontology issue.   It does not matter that the links are not in GO basic.  I really only want to find out why the F-P inferences I expected are missing.

The question is really  why do the
MF annotations to "SNAP receptor activity" not result in a GOC annotation  to "membrane fusion"

 @cmungall  is probably the best person to answer this.  Once I know the reason they are missing  I can figure out  if it a problem with the inference pipeline that will be fixed, or If I need to  fill int he annotation gaps manually. I'm happy to explain the issue on a call sometime  if this does not make sense.
(@cmungall  you only need to consider the comments since 25th Sept - this is where I realise the problem was not what I thought it was originally)

**Comment by pgaudet** [comment]
I think @dougli1sqrd was working on the script that generated those?

**Comment by ValWood** [comment]
@pgaudet  want to revisit this tickdet. It is still an issue for us.

I will summarize
There is a MF-BP  part of link between  SNAP receptor activity and membrane fusion:

<img width="235" alt="Screenshot 2021-04-16 at 15 45 50" src="https://user-images.githubusercontent.com/7359272/115041933-ef483000-9eca-11eb-8012-4a5cc2a8622a.png">

So, I expected the GO  Predictions would include inferred mapping to "membrane fusion" for all SNAP receptor activity annotations.

However
<img width="868" alt="Screenshot 2021-04-16 at 15 49 07" src="https://user-images.githubusercontent.com/7359272/115042350-4a7a2280-9ecb-11eb-9303-ef94080125a4.png">

, you can see that this is not the case:

<img width="779" alt="Screenshot 2021-04-16 at 15 51 04" src="https://user-images.githubusercontent.com/7359272/115042596-8f9e5480-9ecb-11eb-8e83-119d7765809b.png">


@cmungall @dougli1sqrd 
https://www.pombase.org/results/from/id/cd19d97f-3b4b-43b1-8145-aa10a5803f4b

Note that we import these inferences from GOA.

**Comment by ValWood** [comment]
@cmungall  
@pgaudet  understand the issue. We just need to know if the file is correctly generated and if we need to import it separately.

@kimrutherford  do we  expect the  inferred annotations are in GOA, or do we pick up any of these from GO? It might be that we are not getting them because they are not in GOA.
(although we suspect that not all of the annotation expected are generated)

**Comment by ValWood** [comment]
I spoke with @kimrutherford  and we do not get the  inferred annotations from GOA we pick them up from 
  http://snapshot.geneontology.org/products/annotations/pombase-prediction.gaf
which I believe is the correct file.

This means that there are  missing inferences. I suspect that, because the only inferences for "membrane fusion" are IDA, that maybe the pipeline only generates the inferences for experimental annotation.   If the rule is correct it should presumably be applied to all annotation regardless of evidence code?

One of the reasons that these annotations cause problems for curators is that it is really difficult to identify the source.  Especially as they use the same evidence code as the originating annotation. Curators have complained about this a lot over many years. I would love to see a different evidence code used for MP->BP inferences  (IMF - inferred from molecular function?) so that we know immediately how these were derived.

**Comment by ValWood** [comment]
@pgaudet  Did this ever get looked into? We are still not picking up the expected inferences which is a shame...could you bring it up on a future editors call?

**Comment by pgaudet** [comment]
Hi @ValWood 

Chatting with @cmungall @kltm - this code is older, unsupported code that actually needs to be replaced, so we need to prioritize this work with the new code (ontobio) that will be taking care of this. This is underway but needs to be completed. 

Relevant tickets: 
https://github.com/geneontology/pipeline/issues/221
https://github.com/geneontology/pipeline/issues/130

Sorry about that ! 

Pascale

**Comment by ValWood** [comment]
What is the situation on this (just checking, but if it will be a  really long time I might make the annotations manually). I opened the ticket in Sept 2020 ;)

**Comment by ValWood** [comment]
Can we discuss on an editors call in the New Year?

**Comment by ValWood** [comment]
@balhoff  out or date?

**Comment by balhoff** [comment]
@ValWood I'm not sure. It looks like it was decided there wasn't an ontology problem, right? If there is a pipeline task, we should make an issue there.

**Comment by ValWood** [comment]
It's a pipeline task. It's a longstanding issue  propagation of annotation over F-P links does not appear to be  working as expected.

Which is the appropriate tracker to open a ticket?

**Comment by ValWood** [comment]
Here is more evidence that F-P propagations are missing.

<img width="965" alt="Image" src="https://github.com/user-attachments/assets/1a9bcab8-c997-4aad-8b4c-90a092f28445" />

There is an F-P link between transmembrane transporter activity, and transmembrane transport.

This means that ALL transmembrane transporters should be annotated by propagation to "transmembrane transport"

However, the Matrix overlap query shows that they aren't:

95744 in intersection vs 96595 to transmembrane transporter

<img width="965" alt="Image" src="https://github.com/user-attachments/assets/3ce826fb-e47c-4e7e-ab57-27e75a8a2407" />

It would be useful if this was working as expected because presumably we are missing a lot of high quality annotations across all species.

**Comment by pgaudet** [comment]
Hi Val, 

GOA is integrating these annotations. GOC will get them when he pipeline switch is completed. I dont know if you want to leave this open until you can see the results?

**Comment by ValWood** [comment]
Can you tag me in the relevant ticket, and we can close this?

**Comment by pgaudet** [comment]
Hi @ValWood 

The current PomBase annotations  in https://release.geneontology.org/2026-05-19/annotations/index.html 
should implement the interontology links. Can you check whether this is working as expected ?

**Comment by ValWood** [comment]
I don't think so, at least using the same example:


BAsed on

<img width="329" height="506" alt="Image" src="https://github.com/user-attachments/assets/7d496491-2e6a-4832-9420-7f45dd3cf8fc" />

I expected ALL "SNAP receptor activity" to be annotated to "membrane fusion", and they don't appear to be.

<img width="1214" height="150" alt="Image" src="https://github.com/user-attachments/assets/1aae4a42-4427-4b43-9061-1433a314e4cb" />

**Comment by ValWood** [comment]
OK the source evidence has changed. 
I should check with @kimrutherford  that there is no reason that these would not be imported
 
"this is generated by GOA, and given the evidence code 'evidence based on logical inference from manual annotation used in automatic assertion' (ECO:0000364), which maps to IEA
Also, these annotations are assigned by GOC."

**Comment by ValWood** [comment]
~for some reason this pombe is missing this annotation in QuickGO
GO:0005484 | SNAP receptor activity | IBA with S000005562 , S000004016 , S000003180 , S000005378 , PTN000461349 , FBgn0031106 , S000002876 , 621019 , S000000012 , 621005 , S000004795 , O14662 , S000005632 , FBgn0011708 , FBgn0035540 , Q13190 , FBgn0013343 , P56962 , S000006153 , 619747
but we have it in PomBAse. if being present in GOA is a requirement. 
@pgaudet  can you see why this would not be present in QuickGO?~

Not related to this ticket, moved to
https://github.com/geneontology/go-annotation/issues/6453

**Comment by ValWood** [comment]
~conversely, gos1 has this annotation in Quickgo
UniProtKB:Q09835 | gos1 | involved_in | GO:0006906    vesicle fusion | ECO:0000318   IBA | GO_REF:0000033 | PANTHER:PTN000473329 more... | 284812 Schizosaccharomyces pombe (strain 972 / ATCC 24843) | GO_Central
and is therefore annotated to "membrane fusion" in GOA, through its ancestry but this annotation doesn't appear in PomBase
UniProtKB:Q09835	gos1	involved_in	[GO:0006906](https://www.ebi.ac.uk/QuickGO/term/GO:0006906)    vesicle fusion	[ECO:0000318](https://www.ebi.ac.uk/QuickGO/term/ECO:0000318)   IBA	GO_REF:0000033	
PANTHER:PTN000473329 more...
284812 Schizosaccharomyces pombe (strain 972 / ATCC 24843)	GO_Central~

added:
This one is currently filtered from PomBAse becasue this fix has not filtered through
https://github.com/geneontology/go-annotation/issues/6400

**Comment by ValWood** [comment]
for sft1 I do see the new annotation:

UniProtKB:Q09730 | sft1 | involved_in | GO:0061025    membrane fusion | ECO:0000364   IEA | GO_REF:0000108 | GO:0005484 | 284812 Schizosaccharomyces pombe (strain 972 / ATCC 24843) | GOC

but this one doesnt appear in PomBAse.

This does mean that some aspect of the pipeline is working ....
@kimrutherford if this is in GOA (1st June) should we have it by now?


also 
sec20

UniProtKB:O42852 | sec20 | involved_in | GO:0061025    membrane fusion | ECO:0000364   IEA | GO_REF:0000108 | GO:0005484 | 284812 Schizosaccharomyces pombe (strain 972 / ATCC 24843) | GOC

**Comment by ValWood** [comment]
~pep12 is annotated at PomBAse with a TAS
GO:0005484 | SNAP receptor activity | TAS | Takegawa K et al. (2003)
[GO:0005484](https://www.pombase.org/term/GO:0005484)	
[SNAP receptor activity](https://www.pombase.org/term/GO:0005484)
[TAS](http://wiki.geneontology.org/index.php/Traceable_Author_Statement_(TAS))	[Takegawa K et al. (2003)](https://www.pombase.org/reference/PMID:14575697)
but isn't in QuickGO
@pgaudet  do  QuickGO exclude TAS?
This annotation os from 20031027~

Unrelated issue. Moved to a new ticket  TAS annotation in QuickGO geneontology/go-annotation#6454

**Comment by kimrutherford** [comment]
> @kimrutherford if this is in GOA (1st June) should we have it by now?

From a quick look, it's not loaded because the "assigned by" (GOC) is not in our list of assigners to accept.  The list is:
`InterPro,UniProtKB,UniProt,RHEA,IntAct,RNAcentral,CAFA,CACAO`

There are 486 pombe annotations in the GAF file assigned by GOC.  Should we include them in tomorrow's update?

**Comment by ValWood** [comment]
Yes please then I will reevaluate. 
Thanks!

**Comment by kimrutherford** [comment]
PomBase now has those GOC annotations.  255 of the 486 annotations made it through the filtering.  
The annotation of GO:0061025 "membrane fusion" for sft1 mentioned above is here:
https://www.pombase.org/gene/SPAC31A2.13c

Here are the 255 new annotations in GAF format:
https://curation.pombase.org/kmr44/pombase-GOC-IEA-2026-06-11.gaf.tsv

**Comment by ValWood** [comment]
OK so this worked, but I still have 5 SNAP receptors that aren't annotated with membrane fusion:

https://www.pombase.org/results/from/id/028d8b98-6a48-43ca-b5cf-5eda3285f5d5

I'm not sure why. Probably wont be able to look today as I'm on holiday, will look later

**Comment by ValWood** [comment]
It is all working.
The first 2 are missing because of PomBase filters (BP problems on the IBA)
The other 3 are missing because GOA is not importing PomBase TAS or ISS, I opened tickets about that (probably on the wrong tracker though)
